homeassistant/components/pglab/device_sensor.py

```diff
@@ -0,0 +1,56 @@
+"""Device Sensor for PG LAB Electronics."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from pypglab.device import Device as PyPGLabDevice
+from pypglab.sensor import Sensor as PyPGLabSensors
+
+from homeassistant.core import callback
+
+if TYPE_CHECKING:
+    from .entity import PGLabEntity
+
+
+class PGLabDeviceSensor:
+    """Keeps PGLab device sensor update."""
+
+    def __init__(self, pglab_device: PyPGLabDevice) -> None:
+        """Initialize the device sensor."""
+
+        # get a reference of PG Lab device internal sensors state
+        self._sensors: PyPGLabSensors = pglab_device.sensors
+
+        self._ha_sensors: list[PGLabEntity] = []  # list of HA entity sensors
```

Comment:
Storing entities in a common container isn't allowed. Entities is a platform concern. Use an UpdateCoordinator to coordinate updates and tell entities to update state.